lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -509,12 +509,32 @@ export class PublicClientApplication implements IPublicClientApplication {
     /**
      * Returns the signed in account matching username.
      * (the account object is created at the time of successful login)
-     * or null when no state is found
+     * or null when no matching account is found.
+     * This API is provided for convenience but getAccountById should be used for best reliability
      * @returns {@link AccountInfo} - the account object stored in MSAL
      */
     getAccountByUsername(userName: string): AccountInfo|null {
+        if (StringUtils.isEmpty(userName)) {
+            return null;
+        }
+
+        const allAccounts = this.getAllAccounts();
+        return allAccounts && allAccounts.length ? allAccounts.filter(accountObj => accountObj.username.toLowerCase() === userName.toLowerCase())[0] || null : null;
```

Comment:
This line is hard to read, can it be broken up?

Reply:
Broke it up into if/else to make the logic more clear